src/librustdoc/lib.rs

```diff
@@ -683,10 +692,7 @@ type MainResult = Result<(), ErrorGuaranteed>;
 fn wrap_return(diag: &rustc_errors::Handler, res: Result<(), String>) -> MainResult {
     match res {
         Ok(()) => diag.has_errors().map_or(Ok(()), Err),
-        Err(err) => {
-            let reported = diag.struct_err(err).emit();
-            Err(reported)
-        }
+        Err(error) => Err(diag.emit_err(MainError { error })),
```

Comment:
I don't think it makes sense to "translate" this -- it's not any better in this current state, IMO. Can you just add `diagnostic_outside_of_impl` for this function and remove `MainError`?

Ideally we'd change `Result<(), String>` into something that's like `Result<(), T>` where `T: IntoDiagnostic` or something...

Reply:
After some thought, realized that the error string that we got here comes from `Result`, which is in English.

If we would like to make this section translatable, then we need to support translations of `Result::Err`. Is this even feasible?